src/app/data-planes/data/index.ts

```diff
@@ -29,26 +29,28 @@ import { isSet } from '@/utilities/isSet'
 
 export type { TrafficEntry } from './stats'
 
-export type DataplaneInbound = PartialDataplaneInbound & {
+export type Connection = {
+  name: string
+  service: string
+  protocol: string
+}
+export type DataplaneInbound = PartialDataplaneInbound & Connection & {
   health: {
     ready: boolean
   }
-  service: string
   addressPort: string
   serviceAddressPort: string
-  protocol: string
 }
-
-export type DataplaneOutbound = PartialDataplaneOutbound & {
-  service: string
-  protocol: string
+export type DataplaneOutbound = PartialDataplaneOutbound & Connection & {
 }
 
 export type DataplaneGateway = PartialDataplaneGateway & {}
 
 export type DataplaneNetworking = Omit<PartialDataplaneNetworking, 'inbound' | 'outbound'> & {
   inbounds: DataplaneInbound[]
   outbounds: DataplaneOutbound[]
+  inboundName: string
+  type: 'delegated' | 'builtin' | 'standard'
```

Comment:
We use the `type` further below in `networking` to figure out whether we just reshape sidecar inbounds that we are given in the response, or we "make" one for the builtin gateway based on the info we have in networking. So we need to know the `type` here.

Then, instead of figuring it out again for `DataPlaneOverview` we just do:

```
DataplaneOverview.type = DataplaneOverview.DataplaneNetworking.type
```

I would guess at some point in the future we will always access the "networking type" (i.e. sidecar or gateway) from `dataplaneOverview.dataplane.networking.type` and then lose the DataPlaneOverview one, but at the minute that would require more change that isn't required in this PR.

Reply:
I see why that makes it beneficial for the property to be accessible on `networking`.

It be nicer though to have `dataplaneType` on the top-level of a resource.

As a benefit of it being on `networking`, it also becomes available on `Dataplane`s.

Reply:
Not sure if we are on the same page here or not, it sounds like maybe we are, did you notice that currently there is still a `dataplaneType` on the top level resource? So the same value is in two places:

https://github.com/kumahq/kuma-gui/pull/2151/files#diff-6cd4b378952d5211d00c3e499e5d33d9d1c1e40cfcc5e470b4940a5d97eaa8a1R184

I think moving forwards there is a possibility we'll change this so its only accessible from one place `dataplaneOverview.dataplane.networking.type`. I don't really see a reason if a top-level one is just for avoiding typing the word `.networking.` i.e. `dataplaneOverview.dataplane.networking.type` vs `dataplaneOverview.dataplaneType`.

I think I can resolve this one but let me know if not.